src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
```

Comment:
PrimathonAICodeReviewer

The middleware assumes the presence of a JWT token without proper error handling if it's missing or invalid. It's recommended to add validation and error handling for decoding the JWT token.
```diff
+     if (!token) {
+         // Handle case when token is missing
+     }
+     let userDetail = {};
+     try {
+         userDetail = this.jwt.decode(token) || {};
+     } catch (error) {
+         // Handle JWT decoding error
+     }
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->